### PR TITLE
Fixing issue #38

### PR DESCRIPTION
**Pull request by adamdriscoll**

### CODE REVIEW

File: PowerShellTools/DependencyValidator.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Windows;
+using Microsoft.Win32;
+
+namespace PowerShellTools
+{
+    public class DependencyValidator
```
Comment (AndreSayreMSFT) [inline PowerShellTools/DependencyValidator.cs]: Can this be internal?

File: PowerShellTools/DependencyValidator.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        private bool? _previousResult;
+        public bool Validate()
+        {
+            if (_previousResult.HasValue) return _previousResult.Value;
+            return ValidateInstalledPowerShellVersion();
+        }
+
+        public bool ValidateInstalledPowerShellVersion()
+        {
+            if (InstalledPowerShellVersion < new Version(3, 0))
```
Comment (AndreSayreMSFT) [inline PowerShellTools/DependencyValidator.cs]: We don't need to create a new Version instance each validation; we can store it static/readonly :)

File: PowerShellTools/DependencyValidator.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                {
+                    System.Diagnostics.Process.Start("http://go.microsoft.com/fwlink/?LinkID=524571");
+                }
+
+                _previousResult = false;
+                return _previousResult.Value;
+            }
+
+            _previousResult = true;
+            return _previousResult.Value;
```
Comment (AndreSayreMSFT) [inline PowerShellTools/DependencyValidator.cs]: we can just return the previousResult.Value once at the end of this function

File: PowerShellTools/DependencyValidator.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        {
+            if (_previousResult.HasValue) return _previousResult.Value;
+            return ValidateInstalledPowerShellVersion();
+        }
+
+        public bool ValidateInstalledPowerShellVersion()
+        {
+            if (InstalledPowerShellVersion < new Version(3, 0))
+            {
+                if (MessageBox.Show(
```
Comment (AndreSayreMSFT) [inline PowerShellTools/DependencyValidator.cs]: Did we want to check if in automation before popping up a message box?

File: PowerShellTools/DependencyValidator.cs (hunk 1 above)
Comment (adamdriscoll) [inline PowerShellTools/DependencyValidator.cs]: Don't see why not!

File: PowerShellTools/DependencyValidator.cs (hunk 2 above)
Comment (adamdriscoll) [inline PowerShellTools/DependencyValidator.cs]: Agreed.

File: PowerShellTools/DependencyValidator.cs (hunk 4 above)
Comment (adamdriscoll) [inline PowerShellTools/DependencyValidator.cs]: Probably a good idea. Any idea how to check that? I can figure it out but if you know off the top of your head... :D

Comment (adamdriscoll) [inline PowerShellTools/DependencyValidator.cs]: Looks like this is the way to go: https://msdn.microsoft.com/en-us/library/envdte._dte.suppressui.aspx

Comment (AndreSayreMSFT) [inline PowerShellTools/DependencyValidator.cs]: We also have stuff within VsShellUtilities and/or PowerShellTools\Project\SharedProject\Utilities.cs

**Pull request merged**